### ISSUE TITLE
Unexpected Undo Behavior

### ISSUE THREAD

**Issue by ChristopherChudzicki** [SketchResponse/sketchresponse#40]
I'm not sure if the following behavior is intentional. I found it unexpected.

1. Select line-segment tool.
2. Click mouse, drag to draw, release.
3. Click Undo.

I would have expected the "Undo" to completely undo my drawing. Instead, I am left with an initial point for the line segment.

By the way: I realize that line segments can also be drawn by 1) click, 2) release, 3) move mouse, 4) click. Had I drawn the segment by this method, the current behavior is what I would have expected.

Not a big issue, but something I noticed.

**Comment by jefrench** [comment]
I agree. I like the behavior myself. It also makes the undo a true undo of
steps.
